lib/Qandidate/Toggle/ExpressionCondition.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Qandidate\Toggle;
+
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+
+/**
+ * A condition written as a symfony language expression that gets evaluated against the
+ * full context, allowing access to several keys of the context in a single condition
+ */
+class ExpressionCondition extends Condition
+{
+    protected $expression;
+    protected $language;
+
+    /**
+     * @param string             $expression The expression to ve evaluated
+     * @param ExpressionLanguage $language The instance of the Expression Language
+     */
+    function __construct($expression, ExpressionLanguage $language = null)
+    {
+        $this->expression = $expression;
+        $this->language   = $language ?: new ExpressionLanguage();
```

Comment:
Any particular reason for forcing the injection? This kind of abstract the processing of the expression, allowing to change the underlying engine without requiring any change to the user code or even the knowledge of what is going on underneath the library.